internal/service/alertconfiguration/model_alert_configuration.go

```diff
@@ -0,0 +1,333 @@
+				Username:       conversion.StringPtrNullIfEmpty(value.Username),
+				EmailEnabled:   types.BoolValue(value.EmailEnabled != nil && *value.EmailEnabled),
+				SMSEnabled:     types.BoolValue(value.SmsEnabled != nil && *value.SmsEnabled),
+			}
+		}
+		return notifications
+	}
+
+	for i := range n {
+		value := n[i]
+		currState := currStateNotifications[i]
+		newState := TfNotificationModel{
+			TeamName: conversion.StringPtrNullIfEmpty(value.TeamName),
+			Roles:    value.Roles,
+		}
+
+		// sentive attributes do not use value returned from API
+		newState.APIToken = conversion.StringNullIfEmpty(currState.APIToken.ValueString())
+		newState.DatadogAPIKey = conversion.StringNullIfEmpty(currState.DatadogAPIKey.ValueString())
+		newState.OpsGenieAPIKey = conversion.StringNullIfEmpty(currState.OpsGenieAPIKey.ValueString())
+		newState.ServiceKey = conversion.StringNullIfEmpty(currState.ServiceKey.ValueString())
+		newState.VictorOpsAPIKey = conversion.StringNullIfEmpty(currState.VictorOpsAPIKey.ValueString())
+		newState.VictorOpsRoutingKey = conversion.StringNullIfEmpty(currState.VictorOpsRoutingKey.ValueString())
+		newState.WebhookURL = conversion.StringNullIfEmpty(currState.WebhookURL.ValueString())
+		newState.WebhookSecret = conversion.StringNullIfEmpty(currState.WebhookSecret.ValueString())
+		newState.MicrosoftTeamsWebhookURL = conversion.StringNullIfEmpty(currState.MicrosoftTeamsWebhookURL.ValueString())
+
+		// for optional attributes that are not computed we must check if they were previously defined in state
+		if !currState.ChannelName.IsNull() {
+			newState.ChannelName = conversion.StringPtrNullIfEmpty(value.ChannelName)
+		}
+		if !currState.DatadogRegion.IsNull() {
+			newState.DatadogRegion = conversion.StringPtrNullIfEmpty(value.DatadogRegion)
+		}
+		if !currState.EmailAddress.IsNull() {
+			newState.EmailAddress = conversion.StringPtrNullIfEmpty(value.EmailAddress)
+		}
+		if !currState.MobileNumber.IsNull() {
+			newState.MobileNumber = conversion.StringPtrNullIfEmpty(value.MobileNumber)
+		}
+		if !currState.OpsGenieRegion.IsNull() {
+			newState.OpsGenieRegion = conversion.StringPtrNullIfEmpty(value.OpsGenieRegion)
+		}
+		if !currState.TeamID.IsNull() {
+			newState.TeamID = conversion.StringPtrNullIfEmpty(value.TeamId)
+		}
+		if !currState.TypeName.IsNull() {
+			newState.TypeName = conversion.StringPtrNullIfEmpty(value.TypeName)
+		}
+		if !currState.Username.IsNull() {
+			newState.Username = conversion.StringPtrNullIfEmpty(value.Username)
+		}
+
+		newState.NotifierID = types.StringPointerValue(value.NotifierId)
+		newState.IntervalMin = types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.IntervalMin))
+		newState.DelayMin = types.Int64PointerValue(conversion.IntPtrToInt64Ptr(value.DelayMin))
+		newState.EmailEnabled = types.BoolValue(value.EmailEnabled != nil && *value.EmailEnabled)
+		newState.SMSEnabled = types.BoolValue(value.SmsEnabled != nil && *value.SmsEnabled)
+
+		notifications[i] = newState
```

Comment:
we could really do: 

notifications[i] =  TfNotificationModel{
			TeamName: conversion.StringPtrNullIfEmpty(value.TeamName),
			Roles:    value.Roles, ...

no newState needed